src/comgr.cpp

```diff
@@ -1302,7 +1301,11 @@ void BuildHip(const std::string& name,
             opts.push_back("-DCK_AMD_BUFFER_ATOMIC_FADD_RETURNS_FLOAT=1");
 #endif
         opts.push_back("-DHIP_PACKAGE_VERSION_FLAT=" + std::to_string(HIP_PACKAGE_VERSION_FLAT));
-        opts.push_back("-DMIOPEN_DONT_USE_HIP_RUNTIME_HEADERS=1");
+        opts.push_back("-DMIOPEN_DONT_USE_HIP_RUNTIME_HEADERS");
+        /// For now, use only standard <limits> to avoid possibility of
+        /// correctnes or performance regressions.
+        /// \todo Test and enable "custom" local implementation.
+        opts.push_back("-DWORKAROUND_DONT_USE_CUSTOM_LIMITS=1");
```

Comment:
@atamazov it seems that we are ahead of compiler when trying to make changes into mainline.
In ROCm 6.0 dockers:

-- Build with HIP 6.0.23494

In ROCm 6.1 staging docker:

-- Build with HIP 6.0.23464

and very strangely, we should not use custom limits (default here) in the *newer* compiler (first one above), but we still *need* to use it in the older one (second one above).

Reply:
@junliume Are we having problems with the 6.0 release? This should not happen, i.e. it's a bug! What is the target GPU? You may have used  some compilation path that was not tested on my Navi21.

Reply:
It all originated from this familiar issue:
> error: typedef redefinition with different types ('long' vs 'long long')
 
So at least in HIP 6.0.23464 we **cannot** use the standard <limits> header, which we have worked around it by a customized header instead. 
However, it seems that in HIP 6.0.23494 we still **can** use standard <limits> header.
 
It looks to me that some hipRTC changes were **reverted** in ROCm 6.0 branch to keep backward compatibility. However, these changes (reverts) were not propagated to mainline first. Thus we have a newer version of hipRTC with older behavior.
 
**This is problematic to us since we need to identify the applicability of workaround by the compiler/runtime version.** 

In the code we are doing it like:
```
if(HIP_PACKAGE_VERSION_FLAT >= 6000023494ULL || HIP_PACKAGE_VERSION_FLAT <= 6000023494ULL)
                opts.push_back("-DWORKAROUND_DONT_USE_CUSTOM_LIMITS=1");
```

We need the two precise points where this workaround should not be applicable.

Reply:
@junliume We cannot rely on HIP versions reported by staging HIP. **_Those numbers are often very outdated._** Therefore, basically we should not modify our code in order to adapt to the HIP versions from staging.

> In ROCm 6.1 staging docker:
> -- Build with HIP 6.0.23464

This is wrong, because obviously current staging is newer than 6.0 release. Please use `-DMIOPEN_OVERRIDE_HIP_VERSION_MINOR=1` with staging (as described in the PR description), this should help to resolve some problems at least.

Reply:
@junliume And maybe we'll have to enable custom limits with staging HIP, but let's override HIP version first and see what happens.

Reply:
The above staging docker is "MIOpen/CK Staging docker" which is based on the latest mainline :)

Reply:
> The above staging docker is "MIOpen/CK Staging docker" which is based on the latest mainline :)

Yep, QA normally uses HIP Mainline to test MIOpen Staging (and promotes MIOpen to Mainline if passed, which seems reasonable)

Reply:
> @atamazov ...any instructions how to do that? Overriding it ~with env vars~ does not work in my case by adding these to cmake command...

This is what cmake should print, please check:
```bash
# CXX=/opt/rocm/llvm/bin/clang++ cmake ... -DMIOPEN_OVERRIDE_HIP_VERSION_MINOR=1 -DMIOPEN_OVERRIDE_HIP_VERSION_PATCH=0 ...
...
-- Build with HIP 6.0.23494
-- MIOPEN_hip_VERSION_MINOR overriden with 1
-- MIOPEN_hip_VERSION_PATCH overriden with 0
...
```
Then you need to rebuild MIOpen. MIOpen should report HIP version 6.1.0 onto console.